### DISCUSSION TITLE
[Feature request] 关于Docker版本的更新数据库和GeoData导出问题。

### DISCUSSION THREAD

**Discussion by zaynechu** [IrineSistiana/mosdns#728] [Q&A]
### 希望添加的功能

目前在openwrt上的mosdns有更新GeoIP & GeoSite 数据库和GeoData导出并打标签的功能。这两项功能在DNS分流操作上非常实用。
现在，我有一个新思路，就是通过ROS的container功能将mosdns直接部署在主路由上，通过转发标记和mosdns在更底层实现对Domain和IP代理的分流。这样所有的国内流量就可以直接绕过旁路由，从主路由完成访问，最大化拓扑性能。
现在的问题是，我找到所有的mosdns通过ROS docker部署的教程全都需要事先配置静态的GeoData标签集。这样就有了维护上的麻烦。远没有openwrt上的自动化操作来得方便。所以请问没有GUI的docker版本有没有通过配置文件实现相应功能的方式？要如何操作？如果没有能否添加相应功能？谢谢！

**Comment by qingtian110** [comment]
你说的这些，他都不是 mosdns 做的事情，可以说跟 mosdns 一点关系都没有。你说的那些东西是第三方的一个壳

**Comment by zaynechu** [comment]
> 你说的这些，他都不是 mosdns 做的事情，可以说跟 mosdns 一点关系都没有。你说的那些东西是第三方的一个壳

问题就是docker镜像不包含你说的“壳”的功能啊。我也查看了luci-app那边的源码，这里好像涉及到一个v2dat对geoip.dat文件的解包流程。对于ROS这样的高度定制化系统来说，docker外部署几乎是没法完成的，肯定要docker内部署才行。mosdns的wiki上也建议了这套流程。说明作者对此是认可的。所以希望作为新功能直接集成进mosdns的镜像里....

**Comment by qingtian110** [comment]
> > 你说的这些，他都不是 mosdns 做的事情，可以说跟 mosdns 一点关系都没有。你说的那些东西是第三方的一个壳
> 
> 问题就是docker镜像不包含你说的“壳”的功能啊。我也查看了luci-app那边的源码，这里好像涉及到一个v2dat对geoip.dat文件的解包流程。对于ROS这样的高度定制化系统来说，docker外部署几乎是没法完成的，肯定要docker内部署才行。mosdns的wiki上也建议了这套流程。说明作者对此是认可的。所以希望作为新功能直接集成进mosdns的镜像里....

如果官方的docker 包含了“壳”的功能，那才叫见鬼了。

官方的 mosdns 由始至终都只是一个核心，至于怎么使用全靠配置文件。而你说的那些 luci 啥的，它只是帮你写了一份配置。这种是非常第三方的东西，和 mosdns 一点关联都没得。

docker 同样是用户自定义配置，至于怎么设置，怎么添加一些列表，取决于用户怎么去创建容器（-v ）

**Comment by zaynechu** [comment]
> > > 你说的这些，他都不是 mosdns 做的事情，可以说跟 mosdns 一点关系都没有。你说的那些东西是第三方的一个壳
> > 
> > 
> > 问题就是docker镜像不包含你说的“壳”的功能啊。我也查看了luci-app那边的源码，这里好像涉及到一个v2dat对geoip.dat文件的解包流程。对于ROS这样的高度定制化系统来说，docker外部署几乎是没法完成的，肯定要docker内部署才行。mosdns的wiki上也建议了这套流程。说明作者对此是认可的。所以希望作为新功能直接集成进mosdns的镜像里....
> 
> 如果官方的docker 包含了“壳”的功能，那才叫见鬼了。
> 
> 官方的 mosdns 由始至终都只是一个核心，至于怎么使用全靠配置文件。而你说的那些 luci 啥的，它只是帮你写了一份配置。这种是非常第三方的东西，和 mosdns 一点关联都没得。
> 
> docker 同样是用户自定义配置，至于怎么设置，怎么添加一些列表，取决于用户怎么去创建容器（-v ）

好吧，恕我冒昧了... 老实说，我这种纯的业余爱好者要是没有ChatGPT帮忙，估计连什么是GitHub都搞不明白。自定义Docker这种事实在是有点超纲... 除了拜拜各位大佬，也实在是没什么好办法...
